airflow/models.py

```diff
@@ -3511,8 +3515,11 @@ def verify_integrity(self, session=None):
         task_ids = []
         for ti in tis:
             task_ids.append(ti.task_id)
-            if not dag.get_task(ti.task_id) and self.state not in State.unfinished():
-                ti.state = State.REMOVED
+            try:
+                dag.get_task(ti.task_id)
+            except AirflowException:
```

Comment:
So is the guard against it running to protect against this situation:
1. Task starts running
2. While task is running, DAG is modified and task is removed
3. We don't want to interrupt the running task, that's what this guard is for

If so do you mind just commenting the logic to clarify?